styx-common/src/main/java/com/spotify/styx/model/EventVisitor.java

```diff
@@ -37,8 +38,9 @@
   R triggerExecution(@Getter WorkflowInstance workflowInstance, Trigger trigger);
   R info(@Getter WorkflowInstance workflowInstance, Message message);
   R dequeue(@Getter WorkflowInstance workflowInstance);
-  R submit(@Getter WorkflowInstance workflowInstance, ExecutionDescription executionDescription);
-  R submitted(@Getter WorkflowInstance workflowInstance, String executionId);
+  R submit(@Getter WorkflowInstance workflowInstance, ExecutionDescription executionDescription,
+      @Nullable String executionId);
```

Comment:
In `RunSpec` we add `executionId`, yet in `ExecutionDescription` we don't add it but carry it alongside like this... Intended?

I think `ExecutionDescription` should eventually converge with `RunSpec` to be one value class... Any reason not to? Javadocs missing for both.